skfp/model_selection/splitters/maxmin_split.py

```diff
@@ -0,0 +1,311 @@
+
+        train_size : float, default=None
+            The fraction of data to be used for the train subset. If None, it is set
+            to 1 - test_size - valid_size. If valid_size is not provided, train_size
+            is set to 1 - test_size. If train_size, test_size and valid_size aren't
+            set, train_size is set to 0.8.
+
+        valid_size : float, default=None
+            The fraction of data to be used for the test subset. If None, it is set
+            to 1 - train_size - valid_size. If train_size, test_size and valid_size
+            aren't set, train_size is set to 0.1.
+
+        test_size : float, default=None
+            The fraction of data to be used for the validation subset. If None, it is
+            set to 1 - train_size - valid_size. If valid_size is not provided, test_size
+            is set to 1 - train_size. If train_size, test_size and valid_size aren't set,
+            test_size is set to 0.1.
+
+        return_indices : bool, default=False
+            Whether the method should return the input object subsets, i.e. SMILES strings
+            or RDKit `Mol` objects, or only the indices of the subsets instead of the data.
+
+        Returns
+        ----------
+        subsets : tuple[list, list, ...]
+        Tuple with train-test subsets of provided arrays. First two are lists of SMILES strings or RDKit `Mol` objects,
+        depending on the input type. If `return_indices` is True, lists of indices are returned instead of actual data.
+
+        References
+        ----------
+        .. [1] `MaxMain split implementation and its variants
+            https://github.com/deepchem/deepchem_`
+
+        .. [2] `Mark Ashton, John Barnardb, Florence Casset, Michael Charlton, Geoffrey Downsb,
+            Dominique Gorse, JohnHolliday, Roger Lahanac, Peter Willett
+            "Identification of Diverse Database Subsets using Property-Based and Fragment-Based Molecular Descriptions"
+            https://onlinelibrary.wiley.com/doi/10.1002/qsar.200290002_`
+
+        .. [3] `Roger Sayle
+            "Improved RDKit implementation"
+            https://github.com/rdkit/UGM_2017/blob/master/Presentations/Sayle_RDKitDiversity_Berlin17.pdf_`
+
+        .. [4] `Tim Dudgeon
+            "Revisting the MaxMinPicker"
+            https://rdkit.org/docs/cppapi/classRDPickers_1_1MaxMinPicker.html_`
+
+        .. [5] `RDKit MaxMin Picker
+            https://squonk.it/docs/cells/RDKit%20MaxMin%20Picker_`
+    """
+
+    data_size = len(data)
+    train_size, valid_size, test_size = validate_train_valid_test_split_sizes(
+        train_size, valid_size, test_size, len(data)
+    )
+    mols = ensure_mols(data)
+    fps = GetMorganGenerator().GetFingerprints(mols)
+
+    picker = MaxMinPicker()
+    test_idxs = picker.LazyBitVectorPick(
+        fps,
+        poolSize=data_size,
+        pickSize=test_size,
+        seed=random_state,
+    )
+
+    # firstPicks - initial state of picked set
```

Comment:
This picks both train and valid set, right? And you filter it below. It would be nice to write this explicitly here